Add first look guides (#74)
neo4j/docs-ops-manager: modules/ROOT/pages/first-look/docker-first-look.adoc

= Docker first look guide

> :description: this page describes the docker compose provisioning instructions for the NOM server and NOM persistence and how to start agents 

Reviewer: ```suggestion
:description: this page describes the docker compose provisioning instructions for the NOM server and NOM persistence and how to start agents on running Neo4j docker containers.
```

> on running Neo4j docker containers.

Reviewer: ```suggestion

```

Docker compose can be used to provision a set of related containers as services to experiment with or to run in test environments.

> But Docker doesn't recommend using compose for production.

Reviewer: ```suggestion
Docker does not recommend using compose for production.
```

> The manual installation, configuring and running of agents on the Neo4j containers will be lost if the container is stopped. 

Reviewer: ```suggestion
The manual installation, configuring and running of agents on the Neo4j containers is lost if the container is stopped. 
```

== Prerequisites
* unix-based operating system
* Docker
* Docker compose

== Prepare hosts file

> You need to set up the following host names in your /etc/hosts or C:/windows/system32/drivers/etc/hosts file;

Reviewer: ```suggestion
You need to set up the following host names in your `/etc/hosts` or `C:/windows/system32/drivers/etc/hosts file`;
```

[source,, role=noheader]
----
127.0.0.1   storage server db-single
----

==  Create persistence DBMS and NOM server

=== Create an empty directory for self-signed certificates

Create a directory *`~/.nom/ssc`* on your local machine before running Docker compose.
The compose file below specifies that:

* This directory will be mounted into the docker container.
* Self-signed certificates will be generated and saved to the mounted directory during the container start up.
* The generated certificates are used by the NOM server

These certificates are used by the server itself but are also required to configure the NOM agent. 

> NOTE: For *`sudo`* docker compose command, make sure the correct path to *`.nom/ssc`* is set in the bind section below.

Reviewer: ```suggestion
[NOTE]
====
For *`sudo`* docker compose command, make sure the correct path to *`.nom/ssc`* is set in the bind section below.
====
```

      SERVER_SSL_KEY_STORE: file:/certificates/server.pfx
      SERVER_SSL_KEY_STORE_PASSWORD: <SSC_PASSWORD>
      GRPC_SERVER_SECURITY_KEY_STORE_TYPE: PKCS12
      GRPC_SERVER_SECURITY_KEY_STORE: file:/certificates/server.pfx
      GRPC_SERVER_SECURITY_KEY_STORE_PASSWORD: <SSC_PASSWORD>
      CORS_ALLOWEDHEADERS: "*"
      CORS_ALLOWEDORIGINS: "http://localhost:[*],https://localhost:[*], http://server:[*]"
      JWT_SECRET: please-set-a-random-secret-string-here-for-jwt-signing
    volumes:
      - type: bind
        source: ~/.nom/ssc
        target: /certificates
    entrypoint:
      - "sh"
      - "-c"
      - "java -jar app.jar ssc -n server -o /certificates -p <SSC_PASSWORD> -d server -i 127.0.0.1 && java -jar app.jar"
----

Documentation for NOM server docker image is *xref:installation/docker/server.adoc[here]*.

=== Run docker compose 
Run the follwing command:

[source, shell]
----
docker compose -f docker-compose.yaml up <folder for context, typically current folder>
----

=== Open NOM UI
Wait for the server container to start and then go to https://server:8080. 
Login as admin:passw0rd and accept license terms. 

== Set up instance and agent

=== Register agent 

> In NOM UI - navigate to Agent settings (clicking on the sad robot will take you to the correct page) and add a new agent.

Reviewer: ```suggestion
In NOM UI - navigate to Agent settings (clicking on the sad robot takes you to the correct page) and add a new agent.
```

      NEO4J_server_bolt_listen__address: db-single:10001
      NEO4J_server_bolt_advertised__address: db-single:10001
      NEO4J_server_metrics_prometheus_enabled: "true"
      NEO4J_server_metrics_prometheus_endpoint: "localhost:2004"
      NEO4J_server_metrics_filter: "*"
    volumes:
       - type: bind
          source: ~/.nom/ssc
          target: /certificates
    healthcheck:
      test: [ "CMD-SHELL", "echo RETURN 1 | cypher-shell -a bolt://db-single:10001 -u neo4j -p <NEO4J_INSTANCE_PASSWORD> || exit 1" ]
      interval: 10s
      timeout: 10s
      retries: 3
      start_period: 5s
----

=== Run docker compose
Run the follwing command:

[source, shell]
----
docker compose -f docker-compose.instance.yaml up <folder for context, typically current folder>
----

=== Run agent

[source, shell]
----
docker compose -f docker-compose.instance.yaml exec sh -c tar -xvf products/neo4j-ops-manager-agent-*-linux-amd64.tar.gz && neo4j-ops-manager-agent-*/bin/agent console
----

== Explore NOM UI
Go to NOM UI and wait for DBMS to appear - this may take a few minutes. 
You should be able to see that the agent has connected in the agents listing. 

> Once DBMS is shown in home page, double click on the name (initially a generated string) to edit it. 

Reviewer: ```suggestion
Once the DBMS is shown in the home page, double click on the name (initially a generated string) to edit it. 
```
